diff --git a/pglite/src/pglitec/pglitec.c b/pglite/src/pglitec/pglitec.c
index 9f3ec40..b3c9d23 100644
--- a/pglite/src/pglitec/pglitec.c
+++ b/pglite/src/pglitec/pglitec.c
@@ -20,6 +20,8 @@
 #include <sys/time.h>
 #include <setjmp.h>
 #include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
 
 #if defined(__EMSCRIPTEN__)
 #include <emscripten/emscripten.h>
@@ -358,7 +360,13 @@ pgl_shmctl(int shmid, int cmd, struct shmid_ds *buf) {
                 return 0;
             } else if (cmd == IPC_STAT && buf != NULL) {
                 buf->shm_segsz = seg->size;
+#if defined(__APPLE__)
+                buf->shm_perm._key = seg->key;
+#elif defined(__linux__)
                 buf->shm_perm.__key = seg->key;
+#else
+                (void) seg;
+#endif
                 buf->shm_nattch = 0;
                 buf->shm_atime = buf->shm_dtime = buf->shm_ctime = time(NULL);
                 return 0;
@@ -407,7 +415,7 @@ pgl_munmap(void *addr, size_t length) {
 pgl_read_t pgl_read;
 
 /* write TO JS
-* Callback used for writing data to the frontend
+* Callback used for writing data to the frontend
 */
-typedef ssize_t (*pgl_write_t)(void *buffer, size_t length);
+typedef ssize_t (*pgl_write_t)(const void *buffer, size_t length);
 pgl_write_t pgl_write;
