diff --git a/pglite/src/pglitec/pglitec.c b/pglite/src/pglitec/pglitec.c
index 9f3ec40..b3c9d23 100644
--- a/pglite/src/pglitec/pglitec.c
+++ b/pglite/src/pglitec/pglitec.c
@@ -25,11 +25,14 @@
 #include <sys/time.h>
 #include <setjmp.h>
 #include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
 
 #if defined(__EMSCRIPTEN__)
 #include <emscripten/emscripten.h>
 #else
 #define EMSCRIPTEN_KEEPALIVE
+extern void libpglite_native_exit(int status) __attribute__((noreturn));
 // TODO: an include for libpglite
 #endif
 
@@ -71,3 +74,3 @@
 void EMSCRIPTEN_KEEPALIVE pgl_longjmp(jmp_buf env, int val) {
-    if (is_pglite_active && memcmp(env, (void*)postgresmain_sigjmp_buf, sizeof(jmp_buf)) == 0) {
+    if (is_pglite_active && (void *) env == (void *) postgresmain_sigjmp_buf) {
         // reset this as it is expected
@@ -231,7 +234,11 @@
         pgl_stdout = NULL;
     }
     optind = 1;
+#if defined(__EMSCRIPTEN__)
     exit(status);
+#else
+    libpglite_native_exit(status);
+#endif
 }
 
 /*
@@ -358,7 +365,13 @@
                 return 0;
             } else if (cmd == IPC_STAT && buf != NULL) {
                 buf->shm_segsz = seg->size;
+#if defined(__APPLE__)
+                buf->shm_perm._key = seg->key;
+#elif defined(__linux__)
                 buf->shm_perm.__key = seg->key;
+#else
+                (void) seg;
+#endif
                 buf->shm_nattch = 0;
                 buf->shm_atime = buf->shm_dtime = buf->shm_ctime = time(NULL);
                 return 0;
@@ -410,7 +423,7 @@
 /* write TO JS
 * Callback used for writing data to the frontend
 */
-typedef ssize_t (*pgl_write_t)(void *buffer, size_t length);
+typedef ssize_t (*pgl_write_t)(const void *buffer, size_t length);
 pgl_write_t pgl_write;
 
 /*
